src/interpreter.rs

```diff
@@ -67,14 +66,11 @@ impl Execute for Predicate {
                     }
                 }
                 BinaryOperator::Regex => {
-                    let rhs = match &self.rhs {
-                        Value::Regex(r) => r,
-                        _ => unreachable!(),
-                    };
-                    let lhs = match lhs_value {
-                        Value::String(s) => s,
-                        _ => unreachable!(),
-                    };
+                    debug_assert!(lhs_value.my_type() == Type::String);
+                    debug_assert!(self.rhs.my_type() == Type::Regex);
+
+                    let lhs = lhs_value.as_str().unwrap();
+                    let rhs = self.rhs.as_regex().unwrap();
```

Comment:
```suggestion
                    // SAFETY: this only panic if and only if 
                    // the semantic checking didn't
                    // catch the mismatched types,
                    // which is a bug.
                    let lhs = lhs_value.as_str().unwrap();
                    let rhs = self.rhs.as_regex().unwrap();
```

I think a safety comment is enough to explain this case.

Reply:
added.